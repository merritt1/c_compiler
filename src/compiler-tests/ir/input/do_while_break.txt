void main(void) {
	int i;
	i = 3;
	do {
		i = i - i;
		break;
	} while (i);
}