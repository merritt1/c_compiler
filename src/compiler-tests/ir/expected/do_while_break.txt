(procBegin,  main)
(addressOf,     r0,      i)
(constInt,     r1,          3)
(storeWord,     r0,      r1)
(label   ,  _GeneratedLabel_1)
(addressOf,     r2,      i)
(addressOf,     r3,      i)
(addressOf,     r4,      i)
(loadWord,     r5,      r3)
(loadWord,     r6,      r4)
(subSignedWord,     r7,      r5,      r6)
(storeWord,     r2,      r7)
(goto    ,  _GeneratedLabel_3)
(label   ,  _GeneratedLabel_2)
(addressOf,     r8,      i)
(loadWord,     r9,      r8)
(gotoIfTrue,     r9,   _GeneratedLabel_1)
(label   ,  _GeneratedLabel_3)
(procEnd ,  main)
